### ISSUE TITLE
Main menu: some buttons produce too many "clicked" sound events

### ISSUE THREAD

**Issue by akien-mga** [OpenDungeons/OpenDungeons#622]
When you click on some buttons of the main menu (e.g. in the new skirmish UI, when you click on "Start game", or the "Quit" button in the main screen), one can hear the "clicked" sound two or three times in a row. This has no practical incidence on the usability, but maybe there's something to look at?

Some buttons are fine, other are behaving in this funny way.

**Comment by akien-mga** [comment]
I've done some more debugging. Looking at the .layout files, there does not seem to be anything particularly wrong with those buttons. For the "Start game" button, it only produces one "click" sound when you click it without selecting first a level in the skirmish menu. So what produces several "clicked" sound events is actually the transition between the "Select a level" menu and the "Configure seats" menu.

**Comment by hwoarangmy** [comment]
I know that when you click the start button, the level is loaded and the cpu is pretty busy. That may be the reason why there is a glitch sound...

**Comment by Danimal696** [comment]
I discovered clicking on empty ground plays the dig sound.

**Comment by akien-mga** [comment]
> I discovered clicking on empty ground plays the dig sound.

True, I think that's due to a recent change, I don't remember this from earlier versions. Let's open a separate issue for that (#627).

**Comment by Bertram25** [comment]
Adding this to 0.5.0 but as a non-blocker.

**Comment by Bertram25** [comment]
I confirm this is a bug.
I disabled the registration of sound clicks there (by commenting this code):
https://github.com/OpenDungeons/OpenDungeons/blob/development/source/render/Gui.cpp#L92-L99

And I added a direct call of the Gui::playButtonClickSound() function in the Skirmish launch button, as an example, in: bool MenuModeSkirmish::launchSelectedButtonPressed(const CEGUI::EventArgs& e)

The sound is then played only once. I'll check in what EventNameSpace the original events are registered and whether we could limit those to a more specific event.

**Comment by Bertram25** [comment]
I added some logging in the click sound function, and indeed it is called between 5 and 7 times when clicking on the launch button when using the default event registration.

**Comment by Bertram25** [comment]
I added a workaround making it so the sound is only played if not already playing. Even with this, the sounds is sometimes played twice if the app is taking more than half a second to change the menu.

To be perfect here, we might have to reinstate playing a sounds in the button functions. The bug is definitely in the CEGUI::GlobalEventSet object IMHO but I don't think it's easily fixable.

WDYT?

**Comment by oyvindln** [comment]
I think we should investigate further, putting the function call in every button click function is not ideal. It could be that it's only the list box change event that gets called multiple times when launching.

**Comment by akien-mga** [comment]
I don't think it's a CEGUI bug, or maybe more a design flaw. I think I found the culprit: if you comment out those three lines https://github.com/OpenDungeons/OpenDungeons/blob/development/source/render/Gui.cpp#L97-L99 (i.e. sound event on CEGUI::Listbox update), the bug is fixed. But then of course you don't get a sound event when clicking on listboxes.

So I think the issue is that the initialisation of the listboxes in the seat configuration screen triggers the sound event. We need to find a way to initialise those quietly, or to only trigger the click sound on actual user input.

**Comment by Bertram25** [comment]
Cool, then initializing the list could be done in the corresponding menus, right?

**Comment by Bertram25** [comment]
I'll have a look at the available listboxes events and make tries.

**Comment by Bertram25** [comment]
Ok, I've tried ListBox::EventMouseClick and EventMouseDown but this doesn't work for lists.
@oyvindln Would you agree if I add click sounds for selection change in the concerned menus once the list boxes are initialized?
This would fortunately be less of an unoptimization.

Thanks for all the feed-back about that guys. :)

**Comment by oyvindln** [comment]
I guess we don't really have a choice if we want sounds for the action.

**Comment by Bertram25** [comment]
Ok, I'll do that then.
